gateway/ln-gateway/src/lnd.rs

```diff
@@ -510,6 +513,147 @@ impl ILnRpcClient for GatewayLndClient {
+        info!("LND Paying invoice {invoice:?}");
+        let mut client = Self::connect(
+            self.address.clone(),
+            self.tls_cert.clone(),
+            self.macaroon.clone(),
+        )
+        .await?;
+
+        debug!("LND got client to pay invoice {invoice:?}, will check if payment already exists");
+
+        // If the payment exists, that means we've already tried to pay the invoice
+        let preimage: Vec<u8> = if let Some(preimage) = self
+            .lookup_payment(invoice.payment_hash.to_vec(), &mut client)
+            .await?
+        {
+            info!("LND payment already exists for invoice {invoice:?}");
+            bitcoin_hashes::hex::FromHex::from_hex(preimage.as_str()).map_err(|error| {
+                LightningRpcError::FailedPayment {
+                    failure_reason: format!("Failed to convert preimage {error:?}"),
+                }
+            })?
+        } else {
+            // LND API allows fee limits in the `i64` range, but we use `u64` for
+            // max_fee_msat. This means we can only set an enforceable fee limit
+            // between 0 and i64::MAX
+            let fee_limit_msat: i64 =
+                max_fee
+                    .msats
+                    .try_into()
+                    .map_err(|error| LightningRpcError::FailedPayment {
+                        failure_reason: format!(
+                            "max_fee_msat exceeds valid LND fee limit ranges {error:?}"
+                        ),
+                    })?;
+
+            let amt_msat = invoice.amount.msats.try_into().map_err(|error| {
+                LightningRpcError::FailedPayment {
+                    failure_reason: format!("amount exceeds valid LND amount ranges {error:?}"),
+                }
+            })?;
+            let final_cltv_delta = invoice.min_final_cltv_delta.try_into().map_err(|error| {
+                LightningRpcError::FailedPayment {
+                    failure_reason: format!("final cltv delta exceeds valid LND range {error:?}"),
+                }
+            })?;
+            let cltv_limit =
+                max_delay
+                    .try_into()
+                    .map_err(|error| LightningRpcError::FailedPayment {
+                        failure_reason: format!("max delay exceeds valid LND range {error:?}"),
+                    })?;
+
+            debug!("LND payment does not exist for invoice {invoice:?}, will attempt to pay");
+            let payments = client
+                .router()
+                .send_payment_v2(SendPaymentRequest {
+                    amt_msat,
+                    dest: invoice.destination.serialize().to_vec(),
+                    payment_hash: invoice.payment_hash.to_vec(),
+                    route_hints: route_hints_to_lnd(&invoice.route_hints),
+                    final_cltv_delta,
+                    cltv_limit,
+                    allow_self_payment: true,
```

Comment:
Idk if you want this on, this is normally only for when trying to do channel rebalances

Reply:
Idk, I just copied that from the other pay function that argument was already there and seemed to work. Later on, when @m1sterc001guy is back, we need to do some refactoring and go over this code again. I haven't touched the gateway in a while but thought fixing this issue in 0.2 is important enough to try it myself anyway.

Reply:
Do you think it could be harmful?

Reply:
The worst case is that you pay a high fee to route to yourself when it could have been 0